src/d3a/d3a_core/launcher.py

```diff
@@ -56,13 +60,26 @@ def run(self):
             self.job_array = [j for j in self.job_array if j.poll() is None]
 
     def is_crowded(self):
+        self.check_and_wait_for_redis()
         enqueued = self.queue.jobs
         if enqueued:
             earliest = min(job.enqueued_at for job in enqueued)
             if datetime.now()-earliest >= self.max_delay:
                 return True
         return False
 
+    @staticmethod
+    def check_and_wait_for_redis():
+        redis_host = urlparse(REDIS_URL).netloc
+        info = None
+        while not info:
+            try:
+                info = StrictRedis(redis_host).info()
+                logging.debug("REDIS IS ALIVE")
+            except ConnectionError:
+                logging.error("Redis not yet available - sleeping")
+                time.sleep(5)
+
```

Comment:
Since this method is used on both d3a-web and d3a, I think it better belongs to d3a-interface

Reply:
You are right, will do it.